migration/v0.30/apply/20-remove-kube-prometheus-components.sh

```diff
@@ -10,7 +10,7 @@ run() {
   execute)
     for CLUSTER in sc wc; do
       log_info "  - removing kube-state-metrics from $CLUSTER"
-      kubectl_do $CLUSTER delete deployments.apps -l app.kubernetes.io/instance=kube-prometheus-stack,app.kubernetes.io/name=kube-state-metrics --cascade=orphan -n monitoring
+      kubectl_do $CLUSTER delete deployments.apps -l app.kubernetes.io/instance=kube-prometheus-stack,app.kubernetes.io/name=kube-state-metrics,app.kubernetes.io/version!=2.8.0 --cascade=orphan -n monitoring
       log_info "  - removing prometheus-node-exporter from $CLUSTER"
       kubectl_do $CLUSTER delete daemonset -l app=prometheus-node-exporter -n monitoring
```

Comment:
This one as well.

I think I rather would like to see it do a check and then delete, so that the script doesn't print out that it removes something when it doesn't.

Reply:
for the DS the label `app=prometheus-node-exporter` does not exists in the newer version, so I think it is safe to keep it as it is

Reply:
I don't like this as the approach for snippets, as such things are not clear.

I think I would prefer that we have a way to pull the version of installed charts and take actions accordingly. I'm fine with letting it be like this for this release and I'll add functions in the scripts library to do so for next release.

Also the upgrade of kube-prometheus-stack should really all be contained here.

Reply:
I think I fixed everything we discussed. let me know if I missed anything